### ISSUE TITLE
New decoder for system calls

### ISSUE THREAD

**Issue by huseyinsaatci** [cloudflare/ebpf_exporter#209]
I was using ebpf_exporter in my project and taking system call metrics as an output. I wanted to write a custom decoder for system calls. It takes a system call id and converts it to the corresponding system call name, as an example if the id is **3** then the output will be **read**. It is used as:
```
decoders:
  -name: syscall
```
If you consider it worth adding a new decoder for system calls I can open a PR.

**Comment by huseyinsaatci** [comment]
I was using a hardcoded list of system calls. There is an enum list in the syscall package, but it only maps syscalls to syscall numbers, so we cannot reach syscall names from syscall numbers. I do not know if it can be achieved without a hardcoded syscalls list.

**Comment by bobrik** [comment]
Go constants are generated:

* https://github.com/golang/go/blob/go1.20.5/src/syscall/zsysnum_linux_arm64.go#L1-L2

Here's the script that does the generation:

* https://github.com/golang/go/blob/go1.20.5/src/syscall/mksysnum_linux.pl

Perhaps `go generate` can be used to produce syscalls at build time:

* https://go.dev/blog/generate

If not, we can have a `make` target to regenerate syscall tables for `x86_64` and `aarch64` (the two architectures for which we provide `vmlinux.h`). These will be static tables, but at they can be regenerated when needed, same as Go.

**Comment by huseyinsaatci** [comment]
So, I was able to use `go generate`  to generate required syscall lists with these commands:
```
//go:generate sh -c "perl mksysnum_linux.pl /usr/include/asm-generic/unistd.h > syscalls_aarch64.go"
//go:generate sh -c "perl mksysnum_linux.pl /usr/include/asm/unistd_64.h > syscalls_x86_64.go"
```

I edited `mksysnum_linux.pl` script in order to reach the required list. One disadvantage is it is perl dependent, so the user's machine must have perl installed to run the script. 

Data is represented as:
```
var x86_syscalls = map[uint64]string{
	0 : "READ",
	1 : "WRITE",
	2 : "OPEN",
	3 : "CLOSE",
......
}
```
OR
```
var aarch_64_syscalls = map[uint64]string{
	0 : "IO_SETUP",
	1 : "IO_DESTROY",
	2 : "IO_SUBMIT",
	3 : "IO_CANCEL",
......
}
```
I do not know if there is a way to create only one of the files for the target system. 

One question is which file the go generate command should be.
I thought it could be in `byte_order.go` file so the syscall list files can be in util file

I also edited Makefile so when `make build` is run `go generate` is also run .
```
generate:
	go generate ./util
build: generate
.....
```

**Comment by bobrik** [comment]
Could you open a PR? I can provide contextualized feedback there.

**Comment by bobrik** [comment]
Merged #214, thanks!
